contracts/masset/RewardManager.sol

```diff
@@ -137,15 +148,30 @@ contract RewardManager is IRewardManager, Ownable {
 
     /// @notice Send funds from this contract. Only Owner can call.
     /// @param _tokenAddress the token address
-    /// @param _recipient the recipient of thw funds
     /// @param _amount the amount to send
-    function sendFunds(address _tokenAddress, address _recipient, uint256 _amount) external onlyOwner {
+    function extractFunds(address _tokenAddress, uint256 _amount) public onlyOwner {
         require(_tokenAddress != address(0x00), "invalid token address");
-        require(_recipient != address(0x00), "invalid recipient");
-        bool result = IERC20(_tokenAddress).transfer(_recipient, _amount);
+        bool result = IERC20(_tokenAddress).transfer(msg.sender, _amount);
         require(result, "transfer failed");
     }
 
+    /// @notice Send all XUSD funds from this contract to the current reward manager
+    /// Only Masset can call
+    function sendFundsToCurrentRM() public {
+        require(msg.sender == address(masset), "not allowed");
+
+        address currentRMAddress = masset.getRewardManager();
+        if(currentRMAddress == address(this)) {
+            return;
+        }
+        address tokenAddress = masset.getToken();
+        uint256 balance = IERC20(tokenAddress).balanceOf(address(this));
+        if(balance >= DUST_LIMIT) {
+            bool result = IERC20(tokenAddress).transfer(currentRMAddress, balance);
+            require(result, "transfer failed");
+        }
```

Comment:
I don't want to revert the transaction if the amount is ~0. I just don't want to waste gas sending it.

Reply:
but reverts exlicitly indicate that. otherwise there are more efforts required to check if transfer actually happened on each transaction.

Reply:
The point of this if is to prevent it sending 0 funds. That just costs gas and looks silly. This function should NOT revert in this case. If the reward is 0, or very close to it, the function should just return, because that is considered a success. That's the semantics of this function and the way it used in the code.